core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java

```diff
@@ -0,0 +1,92 @@
+package com.baeldung.nullcheck;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.function.Function;
+
+import org.junit.jupiter.api.Test;
+
+class NullValueCheckUnitTest {
+
+    Function<Collection<?>, Long> countNulls = collection -> collection.stream()
+        .filter(Objects::isNull)
+        .count();
+
+    @Test
+    void givenList_whenNullValueAdded_doesNotFail() {
+        // adding nulls to a list
+        Integer[] numberArray = { null, 0, 1, null, 2, 3, null };
+        List<Integer> numbers = Arrays.asList(numberArray);
+        assertEquals(countNulls.apply(numbers), 3);
+
+        // accessing nulls from a list
+        assertNull(numbers.get(0));
+
+        // dereferencing nulls from a list
+        assertThrows(NullPointerException.class, () -> numbers.get(0)
+            .toString());
+    }
+
+    @Test
+    void givenHashSet_whenNullValueAdded_doesNotFail() {
+        // adding nulls to a HashSet
+        Integer[] numberArray = { null, 0, 1, null, 2, 3, null };
+        Set<Integer> numbers = new HashSet<>(Arrays.asList(numberArray));
+        assertEquals(countNulls.apply(numbers), 1);
+
+        // accessing null from a set
+        assertTrue(numbers.contains(null));
+
+        // dereferencing nulls from a set
+        assertThrows(NullPointerException.class, () -> numbers.forEach(item -> item.toString()));
+    }
+
+    @Test
+    void givenTreeSet_whenNullValueAdded_mightFail() {
+        // adding nulls to a TreeSet
+        Integer[] numberArray = { null, 0, 1, null, 2, 3, null };
+        assertThrows(NullPointerException.class, () -> new TreeSet<>(Arrays.asList(numberArray)));
+    }
+
+    @Test
+    void givenHashMap_whenNullKeyValueAdded_doesNotFail() {
+        // adding nulls to key value pairs
+        Integer[] numberArray = { null, 0, 1, null, 2, 3, null };
+        Map<Integer, Integer> numbers = new HashMap<>();
+        Arrays.stream(numberArray)
+            .forEach(integer -> numbers.put(integer, integer));
+        assertEquals(countNulls.apply(numbers.keySet()), 1);
```

Comment:
Wrong order, please swap these 2 arguments: expected and actual value.

Reply:
Fixed.